components/twitch/sources/common.js

```diff
@@ -0,0 +1,43 @@
+const twitch = require("../twitch.app.js");
+
+module.exports = {
+  dedupe: "unique",
+  props: {
+    twitch,
+    db: "$.service.db",
+  },
+  methods: {
+    async paginate(resourceFn, params, type, max = null) {
+      const items = [];
+      let done = false;
+      let count = 0;
+      while (!done) {
+        const { data, pagination } = await resourceFn(params);
```

Comment:
Let me know if I've implemented this correctly or if I need to make changes since this is my first time including retry and backoff logic in a component. Should I be using more of the header info to calculate the backoff?